Add categories bar

fatlotus opened the pull request:
This adds a top bar with all categories, in order, plus a new "free" category.

Test URL: https://add-categories-bar-dot-hosted-caravel.appspot.com

georgeteo commented on caravel/templates/header.html:
```diff
@@ -17,7 +36,7 @@ <h1 class="branding col-xs-9 col-md-2">
   </form>
   <div class="user-button-container col-xs-12 col-md-4">
-    <a href="{{ url_for('new_listing') }}"
+    <a href="{{ url_for('new_listing', q=request.args.get('q')) }}"
```
Add text-decoration: none; color: black

georgeteo commented:
The responsive design for the nav bar jive with the header if the page is small
![screen shot 2015-10-28 at 8 52 37 am](https://cloud.githubusercontent.com/assets/12107160/10790436/51bed834-7d51-11e5-88d7-de962da08598.png)
:

georgeteo commented on caravel/templates/header.html:
```diff
@@ -1,3 +1,22 @@
+  {% for tag, label in categories %}
+    {% if tag in request.args.get('q', '') %}
+    <li class="active">
+      <a href="{{ url_for('search_listings',
```
Add text-decoration: none; color: black?

georgeteo commented on caravel/templates/header.html:
```diff
@@ -1,3 +1,22 @@
+    </li>
+    {% else %}
+    <li>
+      <a href="{{ url_for('search_listings',
```
Same here.

fatlotus commented:
I've changed the styling to black and added some responsiveness. Since the navigation bar shows up as a full menu, I added a JavaScript hack to scroll past it, but there's probably a cleaner way to do it.

fatlotus commented:
Looks good?

georgeteo commented:
Yups. Merge way.
